c++/src/kj/async-io-test.c++

```diff
@@ -300,10 +300,10 @@ TEST(AsyncIo, AncillaryMessageHandlerNoMsg) {
 #endif
 
 // We only support ancillary messages on Unix.
-// MacOS only supports SO_TIMESTAMP on datagram sockets, so this test won't work.
+// MacOS & FreeBSD only support SO_TIMESTAMP on datagram sockets, so this test won't work.
 // Android fails this test for unknown reasons and I don't care because it's an extremely obscure
 // feature anyway.
-#if !_WIN32 && !__CYGWIN__  && !__APPLE__ && !__ANDROID__
+#if !(_WIN32 || __CYGWIN__  || __APPLE__ || __ANDROID__ || __FreeBSD__)
```

Comment:
Nitpick, but I feel that the negation-of-ors is harder to follow than the previous ands-of-negations, can we keep it the old way and just add freeBSD to the end?

Reply:
Done. Your version is 3 character longer. :-P

```
-#if !(_WIN32 || __CYGWIN__  || __APPLE__ || __ANDROID__ || __FreeBSD__)
+#if !_WIN32 && !__CYGWIN__  && !__APPLE__ && !__ANDROID__ && !__FreeBSD__
```